lib/datadog/statsd/sender.rb

```diff
@@ -96,9 +98,21 @@ def start
 
         # initialize a new message queue for the background thread
         @message_queue = @queue_class.new
-        # start background thread
-        @sender_thread = @thread_class.new(&method(:send_loop))
-        @sender_thread.name = "Statsd Sender" unless Gem::Version.new(RUBY_VERSION) < Gem::Version.new('2.3')
+        begin
+          # start background thread
+          @sender_thread = @thread_class.new do
+            begin
+              send_loop
+            ensure
+              @mx.synchronize { @done = true }
+            end
+          end
+          @sender_thread.name = "Statsd Sender" unless Gem::Version.new(RUBY_VERSION) < Gem::Version.new('2.3')
+        rescue ThreadError => e
```

Comment:
I'd suggest adding a test for this, as it's a bit brittle to maintain and understand what led to it. It looks like it'd be easy to do so: the `@thread_class` is already being injected, and thus it can be replaced by something that always raises a dummy `ThreadError` when `#new` is called on it.